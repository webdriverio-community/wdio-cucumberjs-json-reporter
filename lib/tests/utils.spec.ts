import { TEST_SCENARIO_STATS, TEST_SCENARIO_STATS_ERROR } from './__mocks__/mocks';
import { describe, it, expect} from 'vitest'
import Utils from '../utils';

describe( 'utils', () => {
    describe( 'getFailedMessage', () => {
        it( 'should not return an error message if the status is "passed"', () => {
            expect( new Utils().getFailedMessage( TEST_SCENARIO_STATS ) ).toEqual( {} );
        } );

        it( 'should return an error message if the status is "failed"', () => {
            expect( new Utils().getFailedMessage( TEST_SCENARIO_STATS_ERROR ) ).toMatchSnapshot();
        } );
    } );

    describe( 'containsSteps', () => {
        it( 'should return false if none of the values is a step name', () => {
            const steps = [
                { keyword: 'foo' },
                { keyword: 'bar' },
                { keyword: 'foobar' },
            ];
            const language = 'en';
            expect( new Utils().containsSteps( steps, language ) ).toEqual( false );
        } );

        it( 'should return true if step contains the word `Given`', () => {
            const steps = [
                { keyword: 'foo' },
                { keyword: 'bar' },
                { keyword: 'Given' },
            ];
            const language = 'en';
            expect( new Utils().containsSteps( steps, language ) ).toEqual( true );
        } );

        it( 'should return true if step contains the word `When`', () => {
            const steps = [
                { keyword: 'When' },
                { keyword: 'bar' },
                { keyword: 'foo' },
            ];
            const language = 'en';
            expect( new Utils().containsSteps( steps, language ) ).toEqual( true );
        } );

        it( 'should return true if step contains the word `Then`', () => {
            const steps = [
                { keyword: 'bar' },
                { keyword: 'Then' },
                { keyword: 'foo' },
            ];
            const language = 'en';
            expect( new Utils().containsSteps( steps, language ) ).toEqual( true );
        } );

        it( 'should return true if step contains the word `And`', () => {
            const steps = [
                { keyword: 'bar' },
                { keyword: 'And' },
            ];
            const language = 'en';
            expect( new Utils().containsSteps( steps, language ) ).toEqual( true );
        } );
    } );

    describe( 'keywordStartsWith', () => {
        it( 'should return undefined if the first word is not a keyword', () => {
            const language = 'en';

            expect( new Utils().keywordStartsWith( 'Milk is not a reserved keyword', language ) ).toBeUndefined();
        } );

        it( 'should return the first word if it is a keyword', () => {
            const language = 'en';

            // * is a valid keyword in Gherkin
            expect( new Utils().keywordStartsWith( '* I have milk', language ) ).toEqual( '*' );
        } );
    } );
} );
