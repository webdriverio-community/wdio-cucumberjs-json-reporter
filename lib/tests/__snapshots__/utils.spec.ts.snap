// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils getFailedMessage should return an error message if the status is "failed" 1`] = `
{
  "error_message": "Error: expect(received).toEqual(expected) // deep equality

Expected: "bar"
Received: "foo"
    at World.<anonymous> (/Users/wimselles/Git/cucumberjs-json-demo/steps.js:17:19)",
}
`;
